flink-streaming-connectors/flink-connector-hbase/src/main/java/org/apache/flink/streaming/connectors/hbase/HBaseMapper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.hbase;
+
+import org.apache.flink.api.common.functions.Function;
+
+import java.io.Serializable;
+
+/**
+ * Maps a input value to a row in HBase table.
+ *
+ * @param <IN> input type
+ */
+public interface HBaseMapper<IN> extends Function, Serializable {
+
+	/**
+	 * Given an input value return the HBase row key. Row key cannot be null.
+	 *
+	 * @param value
+	 * @return row key
+	 */
+	byte[] rowKey(IN value);
```

Comment:
> In other words, the row key returned from rowKey(IN value) will be passed to createMutations(byte[] rowKey, boolean writeToWAL) method called on the instance returned from actions(IN value). How do you think?

Am not entirely convinced on having a rowKey API and use that in createMutations. Since HBase is going to allow row key as byte[] I think there is no harm in exposing an argument with byte[] which represents the rowkey. Hence i was of that opinion.